Enable GCC/clang -Wshadow compiler flag

droark opened the pull request:
GCC/clang will now throw warnings whenever variables shadow other variables. Armory code was cleaned up to remove these warnings. (Third party code - LMDB and Crypto++ - and code generated by SWIG was not cleaned up.)

droark commented:
Ran CppBlockUtilsTests and DB1kIter. No new tests failed. (Three fail under OS X when running CppBlockUtilsTests.) It's a good idea to double check the code changes, though. I mostly added underscores but there were two variables that were duplicates of the variables they shadowed, with no apparent need for it. I deleted those shadow variables.

goatpig commented:
I have a big commit coming with p2sh + p2wsh wallets with signing, as well as compressed keys. Could you rebase on that after I push that code? It's for tomorrow or the day after I believe. Code is mostly done, writing unit tests and debugging as we speak.

droark commented:
Gaaaaah. Wish I had known this earlier. :/ I'll rebase once it's pushed.

goatpig commented:
Yeah sorry about that, I tend to do few massive pushes lately.

goatpig commented:
The big commit is pushed, go ahead

droark commented:
Rebased and slightly revised. The biggest change is that I went ahead and fixed the warnings in the the LMDB code. This should be fine. However, all these changes need to be carefully considered before accepting them.

goatpig commented:
The changes to the custom lmdb cpp headers are fine, but I can't take any changes to mdb.c. It's basically pointless, as whenever I update LMDB, the entire changeset goes out the toilet.
